### PR TITLE
Ack 1027 commenting components tweaks

### PR DESCRIPTION
**Pull request by roger-in-kiva**
- Commenting components updated to handle data in a more generic way
- ActivityFeedService small tweaks to have a generic method to add child reactions

Screenshot 2024-02-29 at 17 11 16

### CODE REVIEW

File: @kiva/kv-activity-feed/src/index.ts [hunk 1]
```diff
@@ -110,16 +110,17 @@ export default class ActivityFeedService {
...
-	 * Adds a like comment for the current user for the provided activity ID
+	 * Adds a child reaction to a parent for the current user for the provided activity ID
 	 *
-	 * @param commentId The comment ID to add a like
+	 * @param parentId The parent ID to add a child reaction
+	 * @param kind The kind of child reaction
 	 * @returns Whether the like was added successfully
 	 */
-	async addLikeToComment(commentId: string): Promise<boolean> {
+	async addChildReaction(kind: string, parentId: string): Promise<boolean> {
```
Comment (dyersituations) [inline @kiva/kv-activity-feed/src/index.ts:119]: Do we know which `kind` we want to expect for now? If so, let's restrict this further with TS. I don't know the correct strings off-hand, but something like `'comment' | 'like'`.

File: @kiva/kv-components/tests/fixtures/mockFeedActivityData.js [hunk 2]
```diff
@@ -22,8 +22,133 @@ const comments = {
...
+						},
 						parent: '',
-						latest_children: {},
+						latest_children: {
+							like: [
+								{
+									created_at: '2024-02-28T20:04:21.445272Z',
+									updated_at: '2024-02-28T20:04:21.445272Z',
+									id: '290a6c07-143a-416e-b016-ac0d9ac8ad91',
+									user_id: '5314401',
```
Comment (dyersituations) [inline @kiva/kv-components/tests/fixtures/mockFeedActivityData.js:39]: Are these real user IDs in this new data? If so, let's change them to random IDs.

File: @kiva/kv-components/tests/unit/specs/components/KvCommentsAdd.spec.js [hunk 3]
```diff
@@ -63,6 +63,18 @@ describe('KvCommentsAdd', () => {
 		expect(emitted()[ADD_COMMENT_EVENT]).toEqual([[TEST_INPUT]]);
 	});
 
+	it('should emit value when enter key pressed', async () => {
+		const { getByPlaceholderText, emitted } = renderCommentsAdd();
+		const textInput = getByPlaceholderText('Add a comment to this loan...');
```
Comment (dyersituations) [inline @kiva/kv-components/tests/unit/specs/components/KvCommentsAdd.spec.js:68]: Oh random quick change: let's change this text to `Add a comment...`

File: @kiva/kv-activity-feed/src/index.ts (hunk 1 above)
Comment (roger-in-kiva) [reply @kiva/kv-activity-feed/src/index.ts:119]: Tweaky!
Yeah, we are expecting those at the moment, nice call!

File: @kiva/kv-components/tests/fixtures/mockFeedActivityData.js (hunk 2 above)
Comment (roger-in-kiva) [reply @kiva/kv-components/tests/fixtures/mockFeedActivityData.js:39]: We are not using them, so I'll remove.

File: @kiva/kv-components/vue/KvCommentsListItem.vue [hunk 4]
```diff
@@ -96,37 +134,46 @@ export default {
...
 
-		const onClick = (reaction, value) => {
-			if (reaction === REPLY_COMMENT_EVENT) {
-				showInput.value = true;
-				nextTick(() => {
-					commentsAddRef.value.$refs.input.focus();
-				});
-			}
+		const commentText = computed(() => comment?.value?.data?.text ?? '');
+		const authorImage = computed(() => comment?.value?.user?.data?.image ?? '');
```
[the remark this answers is not in the document]
Comment (roger-in-kiva) [reply @kiva/kv-components/vue/KvCommentsListItem.vue:151]: Yeah, nice call about that. Also I didn't notice how do we gonna deal with this at the moment, so just added to have a match with the data mock and show it correctly at the moment, mainly for the styles tweaks.
I'm sure will have a second iteration on this, good you see this!

[the remark this answers is not in the document]
Comment (dyersituations) [reply @kiva/kv-components/vue/KvCommentsListItem.vue:151]: Yeah, for now your integration with CPS can ignore the images.

**Review by dyersituations — APPROVED** [review]

**Pull request merged**